tests/pipeline/test_buyback_auth.py

```diff
@@ -0,0 +1,297 @@
+    def test_compute_cash_buyback_auth(self, dates):
+        self._test_compute(dates)
+
+
+class ShareBuybackAuthLoaderTestCase(EventLoaderCommonTest, TestCase):
+    """
+    Test for share buyback authorizations dataset.
+    """
+    pipeline_columns = {
+        PREVIOUS_BUYBACK_SHARE_COUNT:
+            ShareBuybackAuthorizations.previous_share_count.latest,
+        PREVIOUS_BUYBACK_ANNOUNCEMENT:
+            ShareBuybackAuthorizations.previous_announcement_date.latest,
+        DAYS_SINCE_PREV:
+            BusinessDaysSincePreviousShareBuybackAuth(),
+    }
+
+    @classmethod
+    def setUpClass(cls):
+        cls._cleanup_stack = stack = ExitStack()
+        cls.finder = stack.enter_context(
+            tmp_asset_finder(equities=cls.equity_info),
+        )
+        cls.cols = {}
+        cls.dataset = {sid: df.drop(CASH_FIELD_NAME, 1)
+                       for sid, df in
+                       enumerate(buyback_authorizations)}
+        cls.loader_type = ShareBuybackAuthorizationsLoader
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._cleanup_stack.close()
+
+    def setup(self, dates):
+        zip_with_floats_dates = partial(self.zip_with_floats, dates)
+        num_days_between_dates = partial(self.num_days_between, dates)
+        _expected_previous_buyback_share_count = pd.DataFrame({
+            0: zip_with_floats_dates(
+                ['NaN'] * num_days_between_dates(None, '2014-01-14') +
+                [1] * num_days_between_dates('2014-01-15', '2014-01-19') +
+                [15] * num_days_between_dates('2014-01-20', None)
+            ),
+            1: zip_with_floats_dates(
+                ['NaN'] * num_days_between_dates(None, '2014-01-14') +
+                [13] * num_days_between_dates('2014-01-15', '2014-01-19') +
+                [7] * num_days_between_dates('2014-01-20', None)
+            ),
+            2: zip_with_floats_dates(
+                ['NaN'] * num_days_between_dates(None, '2014-01-09') +
+                [3] * num_days_between_dates('2014-01-10', '2014-01-19') +
+                [1] * num_days_between_dates('2014-01-20', None)
+            ),
+            3: zip_with_floats_dates(
+                ['NaN'] * num_days_between_dates(None, '2014-01-09') +
+                [6] * num_days_between_dates('2014-01-10', '2014-01-14') +
+                [23] * num_days_between_dates('2014-01-15', None)
+            ),
+            4: zip_with_floats_dates(['NaN'] * len(dates)),
+        }, index=dates)
+        self.cols[
+            PREVIOUS_BUYBACK_SHARE_COUNT
+        ] = _expected_previous_buyback_share_count
+        self.cols[PREVIOUS_BUYBACK_ANNOUNCEMENT] = \
+            self.get_expected_previous_event_dates(dates)
+        self.cols[DAYS_SINCE_PREV] = self._compute_busday_offsets(
+            self.cols[PREVIOUS_BUYBACK_ANNOUNCEMENT]
+        )
+
+    @parameterized.expand(param_dates)
+    def test_compute_share_buyback_auth(self, dates):
+        self._test_compute(dates)
```

Comment:
rename `_test_compute` to `test_compute` on the base and then classes will inherit this test when they use the mixin